benchmarks/gluon_bench/benchmarks/expunge.go

```diff
@@ -0,0 +1,163 @@
+package benchmarks
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/bradenaw/juniper/xslices"
+	"github.com/google/uuid"
+	"net"
+
+	"github.com/ProtonMail/gluon/benchmarks/gluon_bench/flags"
+	"github.com/ProtonMail/gluon/benchmarks/gluon_bench/utils"
+	"github.com/emersion/go-imap"
+	"github.com/emersion/go-imap/client"
+)
+
+var (
+	expungeCountFlag    = flag.Uint("expunge-count", 0, "Total number of messages to expunge during expunge benchmarks.")
+	expungeSameMBoxFlag = flag.Bool("expunge-same-mbox", false, "When true run all the expunge test on the same inbox rather than separate ones in parallel.")
+	expungeListFlag     = flag.String("expunge-list", "", "Use a list of predefined sequences to expunge rather than random generated. Only works when -expunge-same-mbox is not set.")
+	expungeAllFlag      = flag.Bool("expunge-all", false, "If set, perform a expunge of the all messages. Only works when -expunge-same-mbox is not set.")
+)
+
+type Expunge struct {
+	seqSets   *ParallelSeqSet
+	mailboxes []string
+}
+
+func NewExpunge() *Expunge {
+	return &Expunge{}
+}
+
+func (*Expunge) Name() string {
+	return "expunge"
+}
+
+func (e *Expunge) Setup(ctx context.Context, addr net.Addr) error {
+	cl, err := utils.NewClient(addr.String())
+	if err != nil {
+		return err
+	}
+
+	defer utils.CloseClient(cl)
+
+	if *expungeSameMBoxFlag {
+		if err := utils.FillBenchmarkSourceMailbox(cl); err != nil {
+			return err
+		}
+
+		status, err := cl.Status(*flags.Mailbox, []imap.StatusItem{imap.StatusMessages})
+		if err != nil {
+			return err
+		}
+
+		messageCount := status.Messages
+
+		if messageCount == 0 {
+			return fmt.Errorf("mailbox '%v' has no messages", *flags.Mailbox)
+		}
+
+		expungeCount := uint32(*expungeCountFlag)
+		if expungeCount == 0 {
+			expungeCount = messageCount / 2
+		}
+
+		seqSets := NewParallelSeqSetExpunge(expungeCount,
+			*flags.ParallelClients,
+			*flags.RandomSeqSetIntervals,
+			*flags.UIDMode)
+
+		if err != nil {
+			return err
+		}
+
+		e.seqSets = seqSets
+		for i := uint(0); i < *flags.ParallelClients; i++ {
+			e.mailboxes = append(e.mailboxes, *flags.Mailbox)
+		}
```

Comment:
```suggestion
		messageCount := status.Messages
		if messageCount == 0 {
			return fmt.Errorf("mailbox '%v' has no messages", *flags.Mailbox)
		}

		expungeCount := uint32(*expungeCountFlag)
		if expungeCount == 0 {
			expungeCount = messageCount / 2
		}

		e.seqSets = NewParallelSeqSetExpunge(
			expungeCount,
			*flags.ParallelClients,
			*flags.RandomSeqSetIntervals,
			*flags.UIDMode,
		)

		for i := uint(0); i < *flags.ParallelClients; i++ {
			e.mailboxes = append(e.mailboxes, *flags.Mailbox)
		}
```

Few things in this code so just making a big suggestion:
- seems there's a useless `if err != nil { ... }` check (prob because `NewParallelSeqSetExpunge` used to return an error?)
- directly assign `e.seqSets` rather than creating a temp variable first (prob left over with the extra error check)
- if you break a function call over multiple lines, it's more idiomatic go style to give each param its own line, followed by the closing `)`

Reply:
Good catch. Addressed.